torch/_dynamo/variables/nn_module.py

```diff
@@ -308,12 +304,21 @@ def named_embed(name, obj):
                         obj,
                         key,
                         name,
-                        source=NNModuleSource(AttrSource(self.source, name)),
+                        source=NNModuleSource(gen_source(self.source, name)),
                         **options,
                     ),
                 ]
             )
 
+        def gen_source(source, name):
+            name_split = name.split(".")
+            if name_split[0] == "":
+                return source
```

Comment:
What if name == "."

Reply:
module name can't be set as "." because we enforce this at https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/module.py#L588.
module name can't be set as "" as well, but the top level module name default is "", so we need this special handling.